### ISSUE TITLE
Some questions about GT

### ISSUE THREAD

**Issue by Fediory** [Anonymous1563/Bayesian-Enhancement-Model#7]
First of all, congratulations to the authors for achieving incredible results in both the LOL and LOLv2 datasets! But I noticed in section 3.3 that it seems to introduce the GT from the test set into the prediction as well. The reason it seems closer to GT is here.
May I ask if this makes sense?

**Comment by BinCVER** [comment]
Hi, Thanks for your interest in our work.

We think that a GT image provided in the test set is very subjective and it is just one possible output. However, we do not want to give up a chance to generate other possible outputs, which may satisfy the aesthetics of different individuals. 
With the full-reference test, we only want to tell whether the BNN can cover the possibility of generating such an image close to the reference image. Since the reference image isn't perfect, a model that matches reference images exactly isn’t necessarily the best. We can only say such a model learned a good mapping from the input and the reference. It could be limiting in real-world applications where users have varied aesthetic preferences, cos such a model might only enhance images in a way that aligns with a possibly flawed reference.

To illustrate this further, we present the following example:
![image](https://github.com/user-attachments/assets/608cf802-c0cf-4092-babb-66d1ed656f26)
Would we say that output B is better than output A only because B is close to the GT?  In terms of brightness preference, Output A might actually be more appealing.

Additionally, we tested using CLIP (with the prompt "[natural, brightness, warm]") as the ranking metric, achieving 25.8 dB (or 28.6 dB with GT_mean) on LOLv1. Using CLIP could allow you to avoid relying on GT images for inference.

**Comment by Fediory** [comment]
I think it's a good idea to use GT to increase model generalizability and image aesthetic quality. I've read similar papers with the same point of view. But it would lead to a non-rigorous experiment in Table1, because you're comparing it to some model that doesn't use GT, then surely your model would have better results. 
So I would suggest, in this experiment, to generate the paired images as if they were unpaired and use the GT mean trick to measure the metrics. on subjective visual comparisons before outputting them using the paired images. 
Would this make more sense?

**Comment by BinCVER** [comment]
We do not use GT for model training or employ any tricks to alter our outputs. In previous DNN methods, it's common to train a model multiple times and report the best results from various checkpoints, often selected based on performance observed on the test set using GT images. Would this be considered GT leakage? If not, similarly, we simply present the best result that our BNN can achieve by comparing multiple outputs with the GT.

In Table 1, our BEM-DNN does not rely on GT for output selection and still achieves better results than previous DNN methods.

**Comment by Fediory** [comment]
Well, a dataset is generally divided into a training set, a test set, and a validation set. We choose the best weights generally on the validation set to ensure the fairness of the experiment. Of course, for a dataset like LOL, which does not have a validation set, there is no guarantee that all previous methods have reasonably divided the validation set. I think this is indeed a drawback of vicious competition in the field, so I would rather recommend the method of averaging the training multiple times, which I've seen in other papers as well.

As well as want to ask if you have manually selected each image in the test set as the closest one based on its GT. If so, then this also looks less reasonable compared to the overall tuning of the original method. Because the overall adjustment can still largely reflect the true generalization ability of a model. For models like BNN that are looking for one picture, multiple generalization results, I think it would be more appropriate to experiment on visual contrast, and some other better metrics, rather than using PSNR SSIM and all that.

**Comment by russellllaputa** [comment]
Since BNN operates differently from deterministic models, a fair comparison based on PSNR and SSIM could only be relevant between deterministic models. We will report BNN's results in a separate section. Additionally, Table 1 also provides the results of our deterministic model, BEM-DNN. **We will bring this to our readers' attention, that any previous methods should only be compared with our BEM-DNN model.** Meanwhile, we will include the following paragraph in our paper to further clarify the evaluation standard:

> For tasks that involve a one-to-many mapping, such as enhancement, full-reference evaluation
provides only a partial assessment of model performance. When there’s a distribution shift between
the training and test sets, the test set results may not fully reflect the model’s ability to generalise
to real-world data. Consequently, no-reference evaluation offers a more meaningful measure of the
model’s practical utility in real-world applications. 

I’ve enjoyed discussing these points with you. Let me know if you have more feedback!

BTW, please consider giving us a star if you think our project is useful!

**Comment by BinCVER** [comment]
I agree that full-reference metrics, like PSNR and SSIM, may not be ideal for assessing BNN’s capabilities. Therefore, we have also included results using non-reference metrics on unpaired datasets, along with visual comparisons.

Since BNN operates differently from deterministic models, a fair comparison based on PSNR and SSIM could only be relevant between deterministic models.  We will report BNN’s results in a separate section. Additionally, Table 1 presents the results of our deterministic model, BEM-DNN. **We will bring this to our readers' attention, that any DNN-based methods should only be compared with our BEM-DNN model.** To further explain the evaluation criteria, we will include the following paragraph in our paper:

> For tasks that involve a one-to-many mapping, such as enhancement, full-reference evaluation provides only a partial assessment of model performance. When there’s a distribution shift between the training and test sets, test set results may not fully reflect the model’s ability to generalize to real-world data. Consequently, no-reference evaluation offers a more meaningful measure of the model’s practical utility in real-world applications.

I’ve enjoyed discussing these points with you, and please feel free to share additional feedback. 
And if you find our project useful, please consider giving us a star!

**Comment by Fediory** [comment]
I am in agreement with your response, as it does make more sense.

One of the reasons I am posting an issue here is because I think your method has some feasibility and research value. Previously I have also looked into how to get a single lowlight map to generate many different enhancements, even including refine more specific structural and detail information, based on the prompt. but for some personal reasons this work had to be aborted.

For this work of yours, I think it's still not novel enough and the details of the study are not clear enough. I think you can continue to REFINE this task to achieve a better visualization, instead of going into a vicious competition on metrics like PSNR like other work.
